src/timetrackings/index.ts

```diff
@@ -1,49 +1,44 @@
 import BaseApi from '../baseApi';
-import { TimeTracking, StartTimeTrackingData, StopTimeTrackingData } from './types';
 import responseHandler from '../utils/responseHandlers';
-import { AxiosRequestConfig } from 'axios';
-import { ReadParams } from '../utils/types';
+import RequestParams from '../utils/requestParams';
+import { TimeTracking, StartTimeTrackingData, StopTimeTrackingData } from './types';
+
 export default class TimeTrackings extends BaseApi {
-  static resourceName = 'timeTrackings';
+  public readonly resourceName = 'timeTrackings';
 
-  public read(options: ReadParams = {}): Promise<TimeTracking[]> {
-    const axiosConfig: AxiosRequestConfig = {
-      params: options,
-    };
-    const response = this._get<TimeTracking>(`${TimeTrackings.resourceName}/read`, axiosConfig);
+  public read(requestParams?: RequestParams<TimeTracking> | Object): Promise<TimeTracking[]> {
+    const params = requestParams instanceof RequestParams ? requestParams?.getParams() : requestParams;
+    const response = this._get<TimeTracking>(`${this.getResourceName()}/read`, { params });
     return responseHandler.requiredList(response);
   }
-  public readById(id: number, options: ReadParams = {}): Promise<TimeTracking[]> {
-    const axiosConfig: AxiosRequestConfig = {
-      params: options,
-    };
-    const response = this._get<TimeTracking>(`${TimeTrackings.resourceName}/read/${id}`, axiosConfig);
+  public readById(id: number, requestParams?: RequestParams<TimeTracking> | Object): Promise<TimeTracking[]> {
+    const params = requestParams instanceof RequestParams ? requestParams?.getParams() : requestParams;
```

Comment:
```suggestion
    const params = requestParams instanceof RequestParams ? requestParams.getParams() : requestParams;
```
I believe requestParams should be narrowed to non-null after the `instanceof`

Reply:
yes you are right, I will remove it